channeldb/invoices.go

```diff
@@ -1913,11 +1918,6 @@ func (d *DB) updateInvoice(hash *lntypes.Hash, refSetID *invpkg.SetID, invoices,
 	// call back.
 	if update.State != nil {
 		setID = update.State.SetID
-	} else if update.SetID != nil {
-		// When we go to cancel HTLCs, there's no new state, but the
-		// set of HTLCs to be cancelled along with the setID affected
-		// will be passed in.
-		setID = (*[32]byte)(update.SetID)
```

Comment:
Need to refresh my memory on AMP, meanwhile it seems the `update.SetID` and `update.State.SetID` are different? Guss my question is, the branch `update.State == nil && update.SetID != nil` doesn't seem to be reflected in `cancelInvoice`.

Reply:
Yes, basically it is how it is set by the callers (the API is not even validated). So when we cancel an htlc for an amp invoice we set it in the top level, while for the rest of operations we set it inside the `State`.

Reply:
actually what is the "top level"? `update.State` or `update.SetID`?

I think with the deletion made in `cancelSingleHtlc`, both the  `update.State` and`update.SetID` will always be nil.

Reply:
oh nvm i saw it's all removed